### DISCUSSION TITLE
[ПЛИ] Кодирование адреса пациента

### DISCUSSION THREAD

**Discussion by ai-fhir** [fhir-ru/core#54] [Административная РГ]
Адрес пациента для протокола лаб исследования требует:

1. Возможность указать `Адрес по месту жительства (постоянной регистрации)` и `Адрес фактического проживания (пребывания)`. FHIR позволяет кодировать тип использования адреса через поле `use` типа `Address` и требует (тип биндинга `Required`) использовать  [свой справочник](https://www.hl7.org/fhir/valueset-address-use.html). Если бы тип поля был `CodeableConcept`, то можно было бы использовать второй код для уточнения, но в нашем случае элемент `use` имеет тип `code`.
1. Возможность указать коды адресов из справочника ГАР (приемник ФИАС)
1. Возможность указать код региона по справочнику НСИ Минздрава (https://nsi.rosminzdrav.ru/#!/refbook)

Пример из документа ЦНИИОИЗ:

```
"address": [
    {
        "line": ["Адрес тесктом"],
        "postalCode": "000000",
        "extension": [
            {
                "url": "https://fias.nalog.ru",
                "extension": [
                    {
                        "url": "AOGUID",
                        "valueUuid": "urn:uuid:316c42e9-3bb0-4b2f-a483-d0d3d9a9ff07"
                    },
                    {
                        "url": "HOUSEGUID",
                        "valueUuid": "urn:uuid:316c42e9-3bb0-4b2f-a483-d0d3d9a9ff07"
                    }
                ]
            },
            {
                "url": "https://nsi.rosminzdrav.ru/#!/refbook",
                "valueCoding": [
                    {
                        "system": "urn:data-instance-id",
                        "value": "urn:oid:1.2.643.5.1.13.13.11.1504",
                        "code": "1",
                        "display": "Адрес по месту жительства (постоянной регистрации)",
                        "version": "1.3"
                    }
                ]
            },
            {
                "url": "https://nsi.rosminzdrav.ru/#!/refbook",
                "valueCoding": [
                    {
                        "system": "urn:data-instance-id",
                        "value": "urn:oid:1.2.643.5.1.13.13.99.2.206",
                        "code": "77",
                        "display": "г. Москва",
                        "version": "6.2"
                    }
                ]
            }
        ]
    },
    {
        "line": ["Адрес тесктом"],
        "postalCode": "000000",
        "extension": [
            {
                "url": "https://fias.nalog.ru",
                "extension": [
                    {
                        "url": "AOGUID",
                        "valueUuid": "urn:uuid:316c42e9-3bb0-4b2f-a483-d0d3d9a9ff07"
                    },
                    {
                        "url": "HOUSEGUID",
                        "valueUuid": "urn:uuid:316c42e9-3bb0-4b2f-a483-d0d3d9a9ff07"
                    }
                ]
            },
            {
                "url": "https://nsi.rosminzdrav.ru/#!/refbook",
                "valueCoding": [
                    {
                        "system": "urn:data-instance-id",
                        "value": "urn:oid:1.2.643.5.1.13.13.11.1504",
                        "code": "3",
                        "display": "Адрес фактического проживания (пребывания)",
                        "version": "1.3"
                    }
                ]
            },
            {
                "url": "https://nsi.rosminzdrav.ru/#!/refbook",
                "valueCoding": [
                    {
                        "system": "urn:data-instance-id",
                        "value": "urn:oid:1.2.643.5.1.13.13.99.2.206",
                        "code": "77",
                        "display": "г. Москва",
                        "version": "6.2"
                    }
                ]
            }
        ]
    }
]
```

Дополнительные предложения:
1. Адрес строкой писать в поле `text` вместо `line`
1. Использовать уникальные url для экстеншнов (+ желательно создать на экстеншны StrucutreDefinition)

**Comment by niquola** [comment]
Можем ли мы для кода региона использовать Address.state? https://github.com/fhir-ru/core/discussions/46

**Reply by maximalexeev** [reply]
Да, это хорошая идея, мы так и делаем. 
Вопрос - как именно передавать. В текущей реализации Нетрики в Address.state передается код региона по справочнику 1.2.643.5.1.13.13.99.2.206

**Comment by niquola** [comment]
Для регистрации и проживания можно попробовать использовать - use=home и use=temp - чтобы не вводить расширение

**Reply by maximalexeev** [reply]
Согласен с предложением, в нашей реализации так и есть, используем use=home и use=temp
home - регистрация, temp - проживание

**Reply by eugenekogan-spb** [reply]
Да, такова реализация Нетрики, и с точки зрения названий это неудачное решение. Название temp пользователи воспринимают как адрес временной регистрации. При передаче в ФРКОР пришлось даже маппить.

**Reply by maximalexeev** [reply]
Название temp пользователи не видят, если уж на то пошло. Или есть системы, в которых **пользователи** рассматривают голый FHIR? 
Подскажите, пожалуйста, более удачный вариант в рамках существующего valueset, позволяющий отразить тип адреса по месту проживания и по регистрации.

**Comment by shamilnn** [comment]
Нужны ли для адреса пациента ещё два extension:
- Patient.address.extension:**preferred** - стандартное FHIR расширение [iso21090-preferred](http://hl7.org/fhir/R4/extension-iso21090-preferred.html)
- Patient.address.extension:**noFixedAddress** - собственное RU Core расширение для бездомных.

**Reply by maximalexeev** [reply]
В своей практике не встречался с необходимостью указывать такие отметки.
